### PR TITLE
IBX-4796: Proper redirect after content publish

### PR DESCRIPTION
**Pull request by barw4**
**JIRA:** https://issues.ibexa.co/browse/IBX-4796

Publishing should redirect to a referrer Location now.

### CODE REVIEW

File: src/lib/Form/Processor/ContentFormProcessor.php [hunk 1]
```diff
@@ -125,10 +126,12 @@ public function processPublish(FormActionEvent $event)
         $event->setPayload('content', $content);
         $event->setPayload('is_new', $draft->contentInfo->isDraft());
 
+        $locationId = !empty($referrerLocation) ? $referrerLocation->id : $content->contentInfo->mainLocationId;
```
Comment (konradoboza) [inline src/lib/Form/Processor/ContentFormProcessor.php]:
```suggestion
        $locationId = $referrerLocation !== null ? $referrerLocation->id : $content->contentInfo->mainLocationId;
```

**Review by konradoboza — APPROVED** [review]

**Review by webhdx — APPROVED** [review]

**Review by ciastektk — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

**Comment by ViniTou** [comment]
It seems, that failing browser test is relevant, could you try to reproduce it in UI?
```
Scenario: Create a folder without a draft                 # vendor/ezsystems/ezplatform-content-forms/features/ContentEdit/create_without_draft.feature:6
    Given that I have permission to create folders          # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::thatIHavePermissionToCreateFolders()
    And there is a Content Type "folder" with the id "1"    # EzSystems\EzPlatformContentForms\Behat\Context\ContentTypeContext::thereIsAContentTypeWithId()
    When I go to "content/create/nodraft/folder/eng-GB/2"   # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::visit()
    Then I should see a folder content edit form            # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::iShouldSeeAContentEditForm()
    When I fill in the folder edit form                     # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::iFillInTheFolderEditForm()
    And I press "Publish"                                   # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::pressButton()
    Then I am on the View of the Content that was published # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::iAmOnTheViewOfTheContentThatWasPublished()
      Behat\Mink\Exception\ElementNotFoundException: Element matching css "span.ezstring-field" not found. in vendor/friends-of-behat/mink/src/WebAssert.php:418
      ```

**Review by ViniTou — COMMENTED** [review]
It seems, that failing browser test is relevant, could you try to reproduce it in UI?
```
Scenario: Create a folder without a draft                 # vendor/ezsystems/ezplatform-content-forms/features/ContentEdit/create_without_draft.feature:6
    Given that I have permission to create folders          # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::thatIHavePermissionToCreateFolders()
    And there is a Content Type "folder" with the id "1"    # EzSystems\EzPlatformContentForms\Behat\Context\ContentTypeContext::thereIsAContentTypeWithId()
    When I go to "content/create/nodraft/folder/eng-GB/2"   # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::visit()
    Then I should see a folder content edit form            # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::iShouldSeeAContentEditForm()
    When I fill in the folder edit form                     # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::iFillInTheFolderEditForm()
    And I press "Publish"                                   # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::pressButton()
    Then I am on the View of the Content that was published # EzSystems\EzPlatformContentForms\Behat\Context\ContentEditContext::iAmOnTheViewOfTheContentThatWasPublished()
      Behat\Mink\Exception\ElementNotFoundException: Element matching css "span.ezstring-field" not found. in vendor/friends-of-behat/mink/src/WebAssert.php:418
      ```

**Review by Nattfarinn — APPROVED** [review]

**Comment by barw4** [comment]
> It seems, that failing browser test is relevant, could you try to reproduce it in UI?

@ViniTou Indeed, a parent Location was resolved as a `referrerLocation` for `nodraft` actions, fixed in https://github.com/ezsystems/ezplatform-content-forms/pull/64/commits/143a0d0f5a6d27af65e8b2c8d95e00ea7414feec.

**Review by konradoboza — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

**Review by webhdx — APPROVED** [review]

**Review by bogusez — APPROVED** [review]

**Pull request merged**